nmos-test.py

```diff
@@ -372,11 +372,15 @@ def run_tests(test, endpoints, test_selection=["all"]):
                 ip_address = endpoints[index]["host"]
             except ValueError:
                 ip_address = socket.gethostbyname(endpoints[index]["host"])
+            try:
+                port = int(endpoints[index]["port"])
+            except ValueError:
+                raise NMOSInitException("Port is missing or is not an integer: {}".format(endpoints[index]["port"]))

```

Comment:
If I enter `foo` in the port text box, I don't see this error, I see "Error: {'endpoints': [{'port': ['Not a valid integer value', 'Please enter a valid port number (0-65535).']}]}", which suggests to me that a string response is getting through the new check above? It only fires for me, if I leave the box empty. (Entering `987654` gets a slightly different JSON message.)

Reply:
The web form does its own validation, but isn't able to protect against missing values as we've had to make the field optional to cope with cases like IS-04-02 where there are multiple boxes.

As this could affect the hostname too, I'll change the method used to catch the error slightly.